microscope/win32.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/python
+"""Win32 specific microscope classes.
+"""
```

Comment:
This whole module fails to load in Linux and so prevents the build of documentation.

Reply:
Probably need to wrap the Win32 imports in a platform check?

Reply:
What if we move the whole win32 stuff into the deviceserver module, and put not all the imports but also the class definitions? The reason to include the class definitions is because they are not usable without the imports. And the reason to put it on the deviceserver module is that its use is limited to that module and we don't want to make it available to users of microscope. And if we are not exposing this to the users, maybe prepend the class names with an underscore.

Reply:
I put it in a separate module so that Win32-only stuff doesn't clutter the platform-agnostic module, but it doesn't have to be like that. The underscores make sense.

Reply:
Put this in a condition so it only declares in a condition.

Reply:
OK - so this can't go into deviceServer: it's meant to be a command line script for Windows service manipulation, but for some reason was missing the if __name__ == '__main__' block; I've added it now.
It seems dumb to wrap the whole thing in a conditional to fix an import on a platform it is not intended to support. What doc tool are you using? Is there another that relies on ast rather than import/inspect?

Reply:
Wait - the command line stuff is all handled in deviceServer, after all, which delegates it to win32. But if we move everything there and wrap it in a platform check, then no docs will be produced for it ... so might as well leave it in its own module and exclude that entire module from the doc generation.